download-logs/download.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+numberregex='^[0-9]+$'
+if ! [[ $1 =~ ${numberregex} && $2 =~ ${numberregex} ]]; then
+    echo "You must specify a start and end time in milliseconds."
+    exit 1
+fi
+
+echo "More specific results can be obtained by providing one or more '--log-stream-name' arguments."
+echo "Log stream names can be determined from CloudWatch in the console."
+echo "(You currently need to modify this script to add stream names.)"
+echo "Are you sure you wish to proceed? [yN]"
+read REPLY && if [[ $REPLY =~ ^[Yy]$ ]]; then
+    aws logs filter-log-events \
+        --start-time $1 \
+        --end-time $2 \
+        --log-group-name "matomo" \
+        --filter-pattern='{ $.status = 5* }' \
+        > aws_logs.txt
+    exit $?
+fi
+exit 1
```

Comment:
Is this really an error/failure? It feels like the script is treating a user's decision to not proceed as a failure, which doesn't seem right. 

As a comparison, `rm -i` prompts user for confirmation to delete, and returns a status of `0` even if the user answers `n`. If the user terminates the command (`ctrl+c`), it treats it as a failure, which makes sense as the command didn't complete.

Reply:
That's a fair point. I based this off a macro I had already written for other things, where not proceeding was meant to fail. But it's not appropriate here.

If we remove this, can we also get rid of the `exit $?` above, do you know? I only added that to prevent failing on exiting the conditional.

Reply:
I've tried doing something similar in a small example, and it looks like that `exit $?` is still needed unless I `set -e` in the script. Any preference as to which approach we use?

Reply:
I'm leaning towards `set -e`.